unstable/wlr-layer-shell-unstable-v1.xml

```diff
@@ -203,21 +204,85 @@
       <arg name="left" type="int"/>
     </request>
 
+    <enum name="keyboard_interactivity" since="4">
+      <description summary="types of keyboard interaction possible for a layer shell surface">
+        Types of keyboard interaction possible for layer shell surfaces. The
+        rationale for this is twofold: (1) some applications are not interested
+        in keyboard events and not allowing them to be focused can improve the
+        desktop experience; (2) some applications will want to take exclusive
+        keyboard focus.
+      </description>
+
+      <entry name="no_focus" value="0">
+        <description summary="no keyboard focus is possible">
+          This value indicates that this surface is not interested in keyboard
+          events and the compositor should never assign it the keyboard focus.
+
+          This is the default value, set for newly created layer shell surfaces.
+
+          This is useful for e.g. desktop widgets that display information or
+          only have interaction with non-keyboard input devices.
+        </description>
+      </entry>
+      <entry name="exclusive" value="1">
+        <description summary="request exclusive keyboard focus">
+          Request exclusive keyboard focus if this surface is above the shell surface layer.
+
+          For layers above the shell surface layers, the seat will always give
```

Comment:
This is the first time this protocol refers to a concept of "shell surface layer", I think? I wouldn't want to force compositors to have a single "shell surface layer" (which contains the regular toplevel surfaces if I understand your intention?).

The existing paragraph about the `layer` enum just describes what a compositor _typically_ does. Maybe a compositor has an always-on-top setting for a toplevel and will put the toplevel between the top and the overlay layer?

Reply:
Maybe instead of "shell surface layer", it could say explicitly that this refers to the top and overlay layers?

Reply:
This sounds good to me.

Reply:
I wonder if there could be more explicit wording (1) specifying whether the compositor is allowed to put non-layer-shell surfaces in any of the layers in the "layer" enum; and (2) noting that compositors will have additional "layers" beside the ones listed in the layer enum, that have a well-defined z-order relative to them (so e.g. if a fullscreen toplevel is above the "top" layer, a layer shell surface in the top layer will not receive exclusive keyboard focus).